### ISSUE TITLE
@aws-cdk/aws-apigateway module: Add ability to reference execution log group ARN or Name

### ISSUE THREAD

**Issue by covertbert** [aws/aws-cdk#15816]
Adding the ability to reference the log group ARN or name that is generated by the `RestApi` construct.

### Use Case

To be able to do *anything* at all with the log group that gets generated (e.g. add a subscription filter) you need to be able to dynamically reference the name or ARN. 

### Proposed Solution

Either:

* Create a readonly property on the instantiated `RestApi` class called `name` or `arn` that can be referenced by other constructs.
* Or, add the ability to set the execution log group name (like you can with access logs) so that you can have a consistent name to reference the log group by.

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by nija-at** [comment]
As far as I am aware, the `RestApi` construct does not create any log groups.
Which logs are you talking about?

**Comment by covertbert** [comment]
@nija-at it does create log groups for *execution* logs. If you want access logs then you create the log group yourself and pass it to RestApi

**Comment by nija-at** [comment]
@covertbert - unfortunately, the CDK does not create these execution logs. They are created automatically by the API Gateway service.

It seems that the log group is always of the format `API-Gateway-Execution-Logs_${RestApiId}/${Stage}`.
We could apply a logic similar to what we do in the `aws-lambda` module using the `LogRetention` construct to get you the execution logs, applied on the `Stage` construct - 

https://github.com/aws/aws-cdk/blob/2023004cc941a0e7a908bf3c90ad9887c6679564/packages/%40aws-cdk/aws-lambda/lib/function.ts#L850-L859.

A workaround is to simply refer the log group by using `LogGroup.fromLogGroupId()` API in the format presented above.

I am unassigning and marking this issue as `p2`, which means that we are unable to work on this immediately.

We use +1s to help prioritize our work, and are happy to revaluate this issue based on community feedback. We accept pull requests if you are interested in contributing this feature.

**Comment by covertbert** [comment]
@nija-at makes sense. The only problem with referencing by ID is that IDs can change when resources are recreated. I was kinda hoping that you'd be able to expose the current ID dynamically. Appreciate this is low priority though, thanks for looking into it.

**Comment by nija-at** [comment]
You should not hardcode the id. Instead, you use [`restApi.restApiId`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-apigateway.RestApi.html#restapiid)

**Comment by covertbert** [comment]
Ahh that makes sense, thanks!

**Comment by shellscape** [comment]
Worth noting for anyone stopping by here, that `LogGroup.fromLogGroupId` is no longer an available static method.

**Comment by shellscape** [comment]
yup

**Comment by rmontoya12** [comment]
It would be nice to specify execution log group.
